addons/web_editor/static/src/js/editor/snippets.editor.js

```diff
@@ -178,6 +178,7 @@ var SnippetEditor = Widget.extend({
         this.templateOptions = templateOptions;
         this.isTargetParentEditable = false;
         this.isTargetMovable = false;
+        this.forgetful = false;
```

Comment:
Your notion of "forgetful" is on the definition of snippet options (as it should). IMO we should find a better name and link it to the "drop" configuration (especially since later, the definition will probably not be in XML anymore). With your implementation, if an element X is targeted by two options:
- One forgetful with data-drop-in="A"
- One non forgetful with data-drop-in="B"

=> The whole snippet editor of X will be considered forgetful and it won't be possible to move it in B (as you change `isTargetMovable`).

What about something like:
- Name it data-drop-as-html / data-initial-drop-only / data-drop-from-panel-only something like that
- Don't have a notion of this at `SnippetEditor` level but only near the uses of data-drop-in / data-drop-near: if it is moving from the right panel, then the "forgetful" options should be considered, otherwise they should not.

In fact, I think it is what you already have if you just remove `this.forgetful` here and just filter `selectorSiblings` and `selectorChildren`